### PR TITLE
Add full support for dlpack API

### LINKED ISSUE #1298: [Bug Report] jax_to_torch uses deprecated dlpack.to() and dlpack.from()
### Describe the bug

The current JaxToTorch wrapper converts jax arrays into a DLPack capsule and then constructs a torch.Tensor from it. This way of transferring tensors is deprecated. Tensors now allow direct conversion via the __dlpack__ protocol.

This is only an issue on newer versions of jax and torch. Updating the wrapper would probably require bumping the version numbers of jax and torch.

### Code example

```shell
import gymnasium as gym
import jax.numpy as jp
import torch


class JaxEnv(gym.Env):
    def __init__(self):
        self.action_space = gym.spaces.Box(low=0, high=1, shape=(1,))
        self.observation_space = gym.spaces.Box(low=0, high=1, shape=(1,))

    def reset(self, seed=None, options=None):
        return jp.array(self.observation_space.sample()), {}

    def step(self, action):
        return jp.array(self.observation_space.sample()), 0, False, False, {}


env = gym.wrappers.JaxToTorch(JaxEnv())
env.step(torch.tensor([0]))  # DeprecationWarning: Calling from_dlpack with a DLPack tensor is deprecated. The argument to from_dlpack should be an array from another framework that implements the __dlpack__ protocol.
```

### System info

- Install via pip
- Gymnasium 1.0.0
- Jax 0.4.38
- torch 2.5.1
- Ubuntu 22.04
- Python 3.11

### Additional context

_No response_

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

### PR DESCRIPTION
**Pull request by amacati**
# Description

This PR adds support for the modern dlpack API to transfer jax arrays to torch tensors and vice-versa. We now determine at initialization if the installed versions of jax and pytorch meet the API requirements, and select the suitable conversion path based on that flag.

Fixes #1298 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by amacati** [comment]
The most problematic issue with the PR is the version check. Versions can be quite complex (see https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers), but I wanted to avoid a dependency on external packages such as [packaging](https://pypi.org/project/packaging/). If you are fine with the additional dependency, this would be the "correct" way of doing it. Otherwise, this implementation may not capture all possible version types. Doing so without additional dependencies would go beyond the scope of this PR.

File: gymnasium/wrappers/jax_to_torch.py [hunk 1]
```diff
@@ -24,6 +24,19 @@
     import jax
     import jax.numpy as jnp
     from jax import dlpack as jax_dlpack
+
+    def _at_least_version(version: str, target: str) -> bool:
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/wrappers/jax_to_torch.py:28]: could we use https://packaging.pypa.io/en/latest/version.html#packaging.version.Version

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Looks good, my only request is to use the packaging versions as this will handle all the version checking complexity with alphas, betas, post, etc versions

File: gymnasium/wrappers/jax_to_torch.py (hunk 1 above)
Comment (amacati) [reply gymnasium/wrappers/jax_to_torch.py:28]: I think we should first discuss the implications of #1298 before adding another dependency. Since parts of the library are broken with older pytorch versions anyways, simply raising the version requirements and getting rid of the flag seems viable again.

File: gymnasium/wrappers/jax_to_torch.py [hunk 2]
```diff
@@ -60,9 +61,7 @@ def _number_torch_to_jax(value: numbers.Number) -> Any:
 @torch_to_jax.register(torch.Tensor)
 def _tensor_torch_to_jax(value: torch.Tensor) -> jax.Array:
     """Converts a PyTorch Tensor into a Jax Array."""
-    tensor = torch_dlpack.to_dlpack(value)  # pyright: ignore[reportPrivateImportUsage]
-    tensor = jax_dlpack.from_dlpack(tensor)  # pyright: ignore[reportPrivateImportUsage]
-    return tensor
+    return jax_dlpack.from_dlpack(value)  # pyright: ignore[reportPrivateImportUsage]
```
Comment (amacati) [inline gymnasium/wrappers/jax_to_torch.py:64]: One more question: Doing this with a tensor that requires gradients throws an error. I think this is reasonable. Tensors that require gradients should not be transferred to another framework, as we have no guarantees on what will happen with the data. Otherwise, we'd have to use `value.detach()` here. Depending on what you prefer, we should either leave it as is, or add the `detach` + tests.

Comment (pseudo-rnd-thoughts) [reply gymnasium/wrappers/jax_to_torch.py:64]: I think we should leave it as is, and if someone has an issue then we can add it

**Review by pseudo-rnd-thoughts — APPROVED** [review]
Thanks for looking at this

**Comment by pseudo-rnd-thoughts** [comment]
Ok, so the new jax requirements doesn't support python 3.8. 
@amacati could you just remove 3.8 from the testing, `.github/workflows/run_pytest.yaml`

**Comment by amacati** [comment]
@pseudo-rnd-thoughts Great, done. I also removed the special requirement for numpy >= 2.0, since it was only required by python 3.8.

**Comment by amacati** [comment]
@pseudo-rnd-thoughts Great, done. I also removed the special requirement for numpy >= 2.0, since it was only required by python 3.8.
